src/routes/create_new_engagement/create_new_engagement.tsx

```diff
@@ -58,24 +60,55 @@ export function CreateNewEngagementForm() {
   const [projectName, setProjectName] = useState(null);
   const [region, setRegion] = useState<string>();
   const [engagementType, setEngagementType] = useState<string>();
+  const [searchTerm, setSearchTerm] = useState<string>();
+  const [copiedEngagement, setCopiedEngagement] = useState<Engagement>();
   const { logEvent } = useAnalytics();
+
+  useEffect(() =>{
+      setCopiedEngagement(findEngagementToCopy(engagements, searchTerm))
+  }, [searchTerm, engagements]);
+
   const submitNewEngagement = async () => {
     if (customerName && projectName) {
-      await createEngagement({
-        customer_name: customerName,
-        project_name: projectName,
-        engagement_region: region,
-        engagement_type:
-          engagementType ??
-          engagementFormConfig?.basic_information?.engagement_type?.options?.find?.(
-            e => e.default
-          )?.value,
-      });
+      if (searchTerm) {
```

Comment:
Could we use a different variable name here? searchTerm doesn't really capture what is happening here